R/bundlePackage.R

```diff
@@ -1,34 +1,17 @@
-bundlePackages <- function(appDir,
+bundlePackages <- function(bundleDir,
                            appMode,
-                           hasParameters = FALSE,
-                           documentsHavePython = FALSE,
-                           quartoInfo = NULL,
+                           extraPackages = character(),
                            verbose = FALSE,
-                           error_call = caller_env()
-                           ) {
-  if (appMode %in% c("static", "tensorflow-saved-model")) {
+                           error_call = caller_env()) {
+  deps <- snapshotRDependencies(bundleDir, extraPackages, verbose = verbose)
+  if (nrow(deps) == 0) {
     return(list())
   }
-
-  # Skip snapshotting R dependencies if an app does not use R. Some
-  # dependencies seem to be found based on the presence of Bioconductor
-  # packages in the user's environment.
-  if (!appUsesR(quartoInfo)) {
-    return(list())
-  }
-
-  # detect dependencies including inferred dependencies
-  inferredRDependencies <- inferRPackageDependencies(
-    appMode = appMode,
-    hasParameters = hasParameters,
-    documentsHavePython = documentsHavePython
-  )
-  deps <- snapshotRDependencies(appDir, inferredRDependencies, verbose = verbose)
   checkBundlePackages(deps, call = error_call)
 
+  copyPackageDescriptions(bundleDir, deps$Package)
   deps$description <- lapply(deps$Package, function(nm) {
     # Remove packageDescription S3 class so jsonlite can serialize
-    # TODO: should we get description from packrat/desc folder?
```

Comment:
We also could update `deps$description` first and then write that information to disk; that would give us one source of truth (the manifest), and one set of calls to `packageDescription`.

Not a firm recommendation.

Reply:
But then you have to round trip from DCF -> list -> DCF and worry about the potential problems that might cause. (Which I don't think are major; it's just not a clear win to me, because you'd still want to carefully test that code.)